cypress/integration/login.spec.js

```diff
@@ -1,152 +1,71 @@
-      .children()
-      .should("have.length", 2)
-      .then(($span) => {
-        expect($span[0].innerText).to.eq(data.errorMsg.emailField);
-      });
-
-    cy.url().should("contain", "/login");
+    login.assertNoMonkeySignEmail(
+      data.invEmail.noDomain,
+      data.userPass,
+      data.errorMsg.emailField
+    );
   });
 
   it("VSL-CRUD06 Login - wrong email - Negative", () => {
-    cy.visit("/");
-    cy.get(loginPage.emailInput).clear().type(data.invEmail.wrongEmail);
-    cy.get(loginPage.passwordInput).type(data.userPass);
-    cy.get(loginPage.logInButton).click();
-
-    cy.get(loginPage.invEmailPassSpan).should(($span) => {
-      expect($span).to.contain(data.errorMsg.oopsMsg);
-    });
-
-    cy.url().should("contain", "/login");
+    login.assertWrongEmail(
+      data.invEmail.wrongEmail,
+      data.userPass,
+      data.errorMsg.oopsMsg
+    );
   });
 
   it("VSL-CRUD07 Login - less then 5 char password - Negative", () => {
-    cy.visit("/");
-    cy.get(loginPage.emailInput).clear().type(data.userEmail);
-    cy.get(loginPage.passwordInput).type(data.invPass.lessCharPass);
-    cy.get(loginPage.logInButton).click();
-
-    cy.get(loginPage.validationSpan)
-      .children()
-      .should("have.length", 2)
-      .then(($span) => {
-        expect($span[1].innerText).to.eq(data.errorMsg.passCharNum);
-      });
-
-    cy.url().should("contain", "/login");
+    login.assertLessCharPass(
+      data.userEmail,
+      data.invPass.lessCharPass,
+      data.errorMsg.passCharNum
+    );
   });
 
   it("VSL-CRUD08 Login - wrong password - Negative", () => {
-    cy.visit("/");
-    cy.get(loginPage.emailInput).clear().type(data.userEmail);
-    cy.get(loginPage.passwordInput).type(data.invPass.wrongPass);
-    cy.get(loginPage.logInButton).click();
-
-    cy.get(loginPage.invEmailPassSpan).should(($span) => {
-      expect($span).to.contain(data.errorMsg.oopsMsg);
-    });
-
-    cy.url().should("contain", "/login");
+    login.assertWrongPass(
+      data.userEmail,
+      data.invPass.wrongPass,
+      data.errorMsg.oopsMsg
+    );
   });
 
-  it("VSL-CRUD08 Login - valid credential - positive", () => {
-    cy.visit("/");
-
-    cy.get(loginPage.emailInput).type(Cypress.env("email"));
-    cy.get(loginPage.passwordInput).type(Cypress.env("password"));
-    cy.get(loginPage.logInButton).click();
-
-    cy.url().should("contain", "/my-organizations");
-
-    cy.get(header.headerTitleSpan)
-      .should("be.visible")
-      .and("contain", data.myOrgHeader);
-
-    cy.get(sidebar.accountImg).should("exist").and("be.visible");
-  });
-
-  it("VSL-CRUD09 Logout", () => {
-    cy.get(sidebar.accountImg).click();
-    cy.get(projectMenu.accountProfileLi).click();
-    cy.get(header.logOutButton).click();
-
-    cy.url().should("contain", "/login");
-
-    cy.get(sidebar.accountImg).should("not.exist");
+  it("VSL-CRUD08 Login & Logout - valid credential - positive", () => {
+    login.loginAndOutUser(
```

Comment:
The naming would have been better if you said something like:
```suggestion
    login.loginThenLogoutTheUser()
```

Reply:
Totally agree. 👍🏻